src/Compilers/CSharp/Portable/BoundTree/BoundExpression.cs

```diff
@@ -739,4 +739,12 @@ internal void VisitAllElements<T>(Action<BoundExpression, T> action, T args)
             }
         }
     }
+
+    internal partial class BoundMethodDefIndex
+    {
+        private partial void Validate()
+        {
+            Debug.Assert(Method.IsDefinition);
```

Comment:
This allows the problem to be detected at an earlier phase. This is also checked in [EmitMethodDefIndexExpression](https://github.com/dotnet/roslyn/blob/9eec48b9253e5c65e22dbe4a41cc30191ee6c974/src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs#L3530).